src/Aspire.Hosting.Azure.Sql/AzureSqlExtensions.cs

```diff
@@ -210,21 +212,24 @@ private static void CreateSqlServer(
         },
         (infrastructure) =>
         {
-            // Creating a new SqlServer instance requires an administrator,
-            // so we need to create one here using the empty PrincipalId/PrincipalName
-            var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
-            infrastructure.Add(principalIdParameter);
-            var principalNameParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalName, typeof(string));
-            infrastructure.Add(principalNameParameter);
+            // Creating a new SqlServer instance requires an administrator. We create a dedicated user assigned managed identity.
+            var adminManagedIdentity = new UserAssignedIdentity("sqlServerAdminManagedIdentity")
+            {
+                Name = Take(Interpolate($"{azureResource.GetBicepIdentifier()}-admin-{GetUniqueString(GetResourceGroup().Id)}"), 63)
+            };
+
+            infrastructure.Add(adminManagedIdentity);
 
             return new SqlServer(infrastructure.AspireResource.GetBicepIdentifier())
             {
                 Administrators = new ServerExternalAdministrator()
                 {
                     AdministratorType = SqlAdministratorType.ActiveDirectory,
                     IsAzureADOnlyAuthenticationEnabled = true,
-                    Sid = principalIdParameter,
-                    Login = principalNameParameter,
+                    Sid = adminManagedIdentity.PrincipalId,
+                    // We can't use adminManagedIdentity.Name as it would end up copying the source expression
+                    // and be converted to a reference() call in ARM which is not supported.
+                    Login = new IdentifierExpression("sqlServerAdminManagedIdentity.name"),
```

Comment:
This should be 

```C#
Login = adminManagedIdentity.Name,
```

Reply:
Have you seen the comment right above the line?

Reply:
@tg-msft - is there a better way to force the expression to reference the property, and not use the property's value?

Maybe: 
```C#
Login = new MemberExpression(new IdentitiferExpression(adminManagedIdentity.BicepIdentifier), "name");
```

Reply:
Something .Compile?

Reply:
@eerhardt with `MemberExpression` it gives `login: 'sqlServerAdminManagedIdentity'.name` :/